### PR TITLE
Add S2259 test cases

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Add test cases from code found on Peach
https://peach.sonarsource.com/project/issues?issues=AYPR_stwHigTOvkdJW4F&open=AYPR_stwHigTOvkdJW4F&id=sharex
https://peach.sonarsource.com/project/issues?issues=AWdBJLiXagf4vZj17Iq5&open=AWdBJLiXagf4vZj17Iq5&id=sharex

### CODE REVIEW

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
It looks like there is a re-base problem. A lot of test code is deleted.

**Comment by pavel-mikula-sonarsource** [comment]
Indeed, a rebase issue

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs [hunk 1]
```diff
@@ -1666,3 +1666,40 @@ public void Nested(object arg)
...
+            value = "";
+        }
+        var isEmpty = string.IsNullOrEmpty(value);
+        if (isEmpty)
+        {
+            value.ToString();   // Compliant
+        }
+        else
+        {
+            value.ToString();   // Compliant
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs:1702]: I don't understand this. On peach we get an FP here, right? If so I would have expected `// Noncompliant FP` here. This doesn't reproduce the FP then. Am I missing something?

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
This should reproduce FPs, right? If so, I would have expected some `// Noncompliant FP` comments.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs:1702]: Exactly, it's a bit weird. There's probably some funky difference in CFG 😒 

I've added these UTs to be sure we don't raise on those under "normal" conditions. And to be able to catch the regressions, if something happens.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs [hunk 2]
```diff
@@ -1666,3 +1666,40 @@ public void Nested(object arg)
...
+        {
+            item = argArray[0];
+        }
+        else
+        {
+            item = null;
+        }
+        if (item != null)
+        {
+            argArray.ToString();    // Compliant
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs:1685]: Same as above

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs (hunk 1 above)
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs:1702]: So we are going to ignore the peach issue, right? And we wait for FP reports from users before taking any further action, right? Does it make sense to leave a comment on peach, that we were not able to reproduce it?

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. You may want to add some comments on peach ("unreproducible" with a link to this PR) in case the issue comes up in the future.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs:1702]: I've added comments, not sure if we will ever notice them :/

**Pull request merged**